src/app/app.component.ts

```diff
@@ -102,10 +91,8 @@ export class AppComponent {
   };
 
   toggleModal = () => {
-    const modalElement: HTMLAbdsModalElement | null = document.querySelector('#modal-1');
-    const currentState = modalElement?.open;
-
-    modalElement?.setAttribute('open', `${!currentState}`);
+    this.isModalOpen = !this.isModalOpen;
+    this.renderer.setAttribute(this.modal.nativeElement, 'open', `${this.isModalOpen ? true : null}`);
```

Comment:
do you need the ternary to be interpolated into a string value?
`${this.isModalOpen ? true : null}`

why not just this.isModalOpen ? 'true' : null

Reply:
kind of hard to tell what I am talking about here, but hopefully this makes sense.

Reply:
Yes, the ternary needs to set the attribute `open` to `null` in order to remove the `open` attribute when the value of `isModalOpen` is `false`, and the third parameter for `this.renderer.setAttribute` must be a string.